godtools/App/Features/Tools/SwiftUI/ToolCategoriesView/ToolCategoryButtonView/ViewModels/BaseToolCategoryButtonViewModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  BaseToolCategoryButtonViewModel.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 6/1/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import SwiftUI
+
+class BaseToolCategoryButtonViewModel: ObservableObject {
+    
+    // MARK: - Properties
+    
+    let categoryText: String
+    var state: ToolCategoryButtonState
+    
+    // MARK: - Published
+    
+    @Published var showBorder: Bool = false
+    @Published var greyOutText: Bool = false
+    
+    // MARK: - Init
+    
+    init(categoryText: String, buttonState: ToolCategoryButtonState) {
+        self.categoryText = categoryText
+        self.state = buttonState
+        
+        setPublishedValues()
+    }
+}
+
+// MARK: - Public
+
+extension BaseToolCategoryButtonViewModel {
+    func setPublishedValues() {
```

Comment:
Hey @rachaelblue it seems this is supposed to be invoked whenever ```state: ToolCategoryButtonState``` is mutated.  However, I would be able to mutate that property without this method ever getting called.  A safer way would be to force a state argument here, maybe instead of ```setPublishedValues()``` method it could be ```func setButtonState(state: ToolCategoryButtonState)``` that way state is forced here and those published values are updated when the state is intended to be changed.

Reply:
That makes sense, good catch!